### PR TITLE
test: unit tests for backend pure logic (cache, catalog, bedrock)

### LINKED ISSUE #816: test: unit tests for backend pure logic (cache, adapters/catalog, bedrock)
Add unit tests for backend modules that contain pure logic testable without any model or credentials.

## Scope

| File | Stmts | What's missing | Est. tests |
|------|-------|---------------|------------|
| \`cache.py\` | 28 | LRU eviction, \`on_evict\` callback, capacity edge cases — zero dedicated tests despite being used by HF backend | ~8 |
| \`adapters/catalog.py\` | 28 | Metadata lookup, \`ValueError\` on unknown name, \`known_intrinsic_names()\` — pure dict/Pydantic logic | ~5 |
| \`bedrock.py\` | 42 | URI construction, \`ModelIdentifier\` matching, env var validation — testable without AWS credentials | ~5 |

## Notes

- All three files are small and self-contained — natural "misc backends" grouping
- \`cache.py\` is the most valuable: it's a pure data structure (LRU with eviction callbacks) used in production but never directly tested
- Part of #813

~18 tests, quick to write and review.

### PR DESCRIPTION
**Pull request by planetf1**
# test: unit tests for backend pure logic (cache, catalog, bedrock)

## Type of PR

- [ ] Bug Fix
- [ ] New Feature
- [ ] Documentation
- [x] Other

## Description
- [x] Link to Issue: Fixes #816

26 unit tests across 3 files for `backends/cache`, `backends/adapters/catalog`, and `backends/bedrock`. All pure logic — no model, no credentials, no network. Runs in ~7s.

### Why these tests exist

All three modules contain self-contained logic (LRU eviction, catalogue lookup, URI construction, pattern matching) that had little or no dedicated unit coverage. `cache.py` in particular is a production data structure used by the HF backend but was never directly tested. Part of the broader #726 testing epic.

### Coverage improvement

| File | Stmts | Before | After |
|------|-------|--------|-------|
| `cache.py` | 34 | 36% | **100%** |
| `adapters/catalog.py` | 22 | 88% | **100%** |
| `bedrock.py` | 42 | 19% | **87%** |

`bedrock.py` caps at 87% because the remaining lines are `list_mantle_models` and `stringify_mantle_model_ids` — thin wrappers around the OpenAI client hitting AWS. Those are correctly covered by the existing e2e test (`test_bedrock.py`, gated behind `require_api_key`).

### Test breakdown

| File | Tests | What's covered |
|------|-------|----------------|
| `test_cache.py` | 11 | put/get contract, LRU eviction order, `on_evict` callback (single and multiple), get-promotes-to-MRU, update-existing-key without eviction, zero capacity no-op, capacity-one boundary |
| `test_adapters/test_catalog.py` | 7 | `known_intrinsic_names()` enumeration, `fetch_intrinsic_metadata()` valid/invalid lookup, `ValueError` on unknown name, defensive copy verification, adapter type defaults vs LORA-only entries |
| `test_bedrock_unit.py` | 8 | `_make_region_for_uri` default, `_make_mantle_uri` construction (default + custom region), `ModelIdentifier` without `bedrock_name` raises, missing env var assertion, happy-path backend creation with `ModelIdentifier` and raw `str` (mocking only `list_mantle_models` at the network boundary), model-not-in-region error |

### Design decisions

- `cache.py`: No mocking needed — pure data structure tests with eviction callbacks tracked via a plain list
- `catalog.py`: Tests against the real catalogue data (not mocked entries) to catch accidental deletions or structural changes
- `bedrock.py`: Happy-path tests mock only `list_mantle_models` (the sole network call) with a lightweight `_FakeModel` dataclass, so the full pattern matching, env var validation, URI construction, and `OpenAIBackend` instantiation all run for real

### Testing
- [x] Tests added to the respective file if code was changed
- [x] New code has 100% coverage if code was added
- [ ] Ensure existing tests and github automation passes (a maintainer will kick off the github automation when the rest of the PR is populated)

### CODE REVIEW

File: test/backends/test_cache.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    assert cache.get("c") == 3
+
+
+def test_on_evict_callback_receives_evicted_value():
+    evicted: list[str] = []
+    cache = SimpleLRUCache(capacity=2, on_evict=evicted.append)
+    cache.put("a", "val_a")
+    cache.put("b", "val_b")
+    cache.put("c", "val_c")  # evicts "a"
+    assert evicted == ["val_a"]
```
Comment (markstur) [inline test/backends/test_cache.py:65]: I don't see a cache size test after eviction.  It might be overkill to add a size check to all the tests, but if I was only looking at one like this I'd probably test size.

**Review by markstur — APPROVED** [review]
I didn't finish comparing coverage numbers with yours (still running).
But lgtm anyway.  Let's merge it.
See one comment.
